src/driver/Firestore/InProcessFirestore.ts

```diff
@@ -1226,3 +1230,189 @@ class InProcessFirestoreWriteBatch implements IFirestoreWriteBatch {
         }
     }
 }
+
+/*
+    Simple in-process implementation of Firestore BulkWriter
+    https://googleapis.dev/nodejs/firestore/latest/BulkWriter.html
+*/
+class InProcessFirestoreBulkWriter implements IFirestoreBulkWriter {
+    private readonly writeOperations: Array<BulkWriterOperation> = []
+
+    private closed = false
+
+    // Commits all enqueued writes and marks the BulkWriter instance as closed.
+    // https://googleapis.dev/nodejs/firestore/latest/BulkWriter.html#close
+    async close(): Promise<void> {
+        await this.flush()
+        this.closed = true
+    }
+
+    // Create a document with the provided data. This single operation will fail if a document exists at its location.
+    // https://googleapis.dev/nodejs/firestore/latest/BulkWriter.html#create
+    create(documentRef: IFirestoreDocRef<IFirestoreDocumentData>, data: IFirestoreDocumentData): Promise<IFirestoreWriteResult> {
+        this.throwIfClosed()
+
+        const createPromise = 
+            async () => {
+                if ((await documentRef.get()).exists) {
+                    throw new FirestoreError(
+                        GRPCStatusCode.ALREADY_EXISTS,
+                        `Document already exists: ${documentRef.path}`,
+                    )
+                }
+                await documentRef.set(data)
+            }
+        const bulkWriteOpPromise = this.enqueue(createPromise)
+        silencePromise(bulkWriteOpPromise)
+        return bulkWriteOpPromise
+    }
+
+    delete(documentRef: IFirestoreDocRef<IFirestoreDocumentData>, precondition?: IPrecondition): Promise<IFirestoreWriteResult> {
+        this.throwIfClosed()
+        if(precondition) {
+            throw new Error(
+                "InProcessFirestorBulkWriter.delete with precondition is not implemented",
+            )
+        }
+
+        const bulkWriteOpPromise = this.enqueue(async () => await documentRef.delete())
+        silencePromise(bulkWriteOpPromise)
+        return bulkWriteOpPromise
+    }
+
+    // Commits all writes that have been enqueued up to this point in parallel.
+    // Returns a Promise that resolves when all currently queued operations have been committed. 
+    // https://googleapis.dev/nodejs/firestore/latest/BulkWriter.html#flush
+    async flush(): Promise<void> {
+        this.throwIfClosed()
+
+        await Promise.all(this.writeOperations.map(async (writeOperation) => {
```

Comment:
You may want to limit how many operations are performed in parallel. 20-100 will be okay, but 1000s might cause memory issues in tests.